File: app/src/main/java/yoyo/app/android/com/yoyoapp/Trip/booking/BookingRepository.java
package yoyo.app.android.com.yoyoapp.Trip.booking;

import android.content.Context;
import androidx.lifecycle.MutableLiveData;
import org.json.JSONObject;
import yoyo.app.android.com.yoyoapp.Trip.ApiService;

public class BookingRepository {
    private ApiService apiService;
    private static BookingRepository instance;
    private MutableLiveData<String> bookingIdMutableLiveData;
    private Context context;

    public static BookingRepository getInstance(Context context)
    {
        if (instance == null) {
            instance = new BookingRepository(context);
        }
        return instance;
    }

    private BookingRepository(Context context)
    {
        this.context = context;
        apiService = new ApiService(context);
    }

    public MutableLiveData<String> getBookingId(String scheduleId, JSONObject jsonObject) {
        bookingIdMutableLiveData = new MutableLiveData<>();
        setBooking(scheduleId,jsonObject);
        return bookingIdMutableLiveData;
    }

    private void setBooking(String scheduleId, JSONObject jsonObject) {
        apiService.sendBookingRequest(scheduleId,jsonObject,bookingId ->{
            bookingIdMutableLiveData.postValue(bookingId);
        });
    }
}

File: app/src/main/java/yoyo/app/android/com/yoyoapp/trip/booking/firstPage/BookingRepository.kt
package yoyo.app.android.com.yoyoapp.trip.booking.firstPage

import android.content.Context
import androidx.lifecycle.MutableLiveData
import org.json.JSONObject
import yoyo.app.android.com.yoyoapp.DataModels.TourRequest
import yoyo.app.android.com.yoyoapp.DataModels.Traveller
import yoyo.app.android.com.yoyoapp.trip.ApiService
import yoyo.app.android.com.yoyoapp.trip.ApiService2
import yoyo.app.android.com.yoyoapp.trip.api.BookScheduleRequest
import yoyo.app.android.com.yoyoapp.trip.api.BookScheduleResponse

class BookingRepository (context: Context) {
    private val apiService: ApiService = ApiService(context)
    private val apiService2: ApiService2 = ApiService2(context)

    fun getCompanionTraveller(f: (ArrayList<Traveller>) -> Unit) {
        apiService.getTravellersCompanionsRequest { f(it)}
    }

    fun bookSchedule(scheduleId: String,request: BookScheduleRequest, f: (BookScheduleResponse) -> Unit){
        apiService2.sendBookingRequest(scheduleId,request)
        {
            f(it)
        }
    }
}
